pipelines/data-pipeline/build.yaml

```diff
@@ -0,0 +1,124 @@
+pool:
+  vmImage: 'ubuntu-latest'
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+      - tech-debt/new-pipelines
+  paths:
+    include:
+      - data-pipeline
+      - pipelines/data-pipeline
+
+pr:
+  paths:
+    include:
+      - data-pipeline
+
+variables:
+  Version.Revision: $[counter(variables['Build.SourceBranchName'], 0)]
+  Version.BuildNumber: 0.0.$(Version.Revision)
+  Version.BuildVersion: $(Version.BuildNumber)
+  TerraformFolder: 'data-pipeline/terraform'
+  TerraformDirectory: '$(System.DefaultWorkingDirectory)/$(TerraformFolder)'
+  ShouldDeploy: $[and(or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),eq(variables['Build.SourceBranch'], 'refs/heads/tech-debt/new-pipelines')), ne(variables['Build.Reason'], 'PullRequest'))]
+
+stages:
+  - stage: Build
+    displayName: 'Build'
+    jobs:
+      - job: SetBuildNumber
+        displayName: 'Set build number'
+        steps:
+          - checkout: none
+
+          - bash: |
+              echo "##vso[task.setvariable variable=Version.BuildNumber]$(Version.BuildNumber)-prerelease"
+              echo "##vso[task.setvariable variable=Version.BuildVersion]$(Version.BuildNumber)"
+              echo "##vso[build.updatebuildnumber]$(Version.BuildNumber)-prerelease"
+            condition: or(eq(variables['Build.SourceBranch'], 'refs/heads/main'), eq(variables['Build.Reason'], 'PullRequest'))
+            displayName: 'Set pre-release build number'
+
+      - job: FmtValidateTf
+        displayName: "Lint and validate terraform"
+        steps:
+          - template: ..\common\fmt-validate-terraform.yaml
+            parameters:
+              TerraformDirectory: $(TerraformDirectory)
+
+      - job: RunCheckOv
+        displayName: "Static analysis terraform"
+        steps:
+          - template: ..\common\scan-terraform.yaml
+            parameters:
+              TerraformDirectory: $(TerraformDirectory)
+
+      - job: ZipStaticArtifacts
+        displayName: 'Zip static artifacts'
+        condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+        steps:
+          - template: ..\common\publish-terraform-artifacts.yaml
+            parameters:
+              TerraformFolder: $(TerraformFolder)
+              Module: 'data-pipeline'
+
+      - job: RunTests
+        displayName: 'Run unit tests'
+        steps:
+          - template: steps\run-unit-tests.yaml
+
+      - job: BuildAndPush
+        dependsOn: [ RunTests ]
+        displayName: 'Build and push image'
```

Comment:
Do we want the image pushing on PR builds?

Reply:
Pending question around how we want to test PR's, with end 2 end or at least smoke tests for the pipeline.